pkg/apis/serving/v1alpha1/revision_defaults.go

```diff
@@ -21,38 +21,28 @@ import (
 
 	"github.com/knative/pkg/apis"
 	corev1 "k8s.io/api/core/v1"
-
-	"github.com/knative/serving/pkg/apis/config"
 )
 
 func (r *Revision) SetDefaults(ctx context.Context) {
 	r.Spec.SetDefaults(apis.WithinSpec(ctx))
 }
 
 func (rs *RevisionSpec) SetDefaults(ctx context.Context) {
-	cfg := config.FromContextOrDefaults(ctx)
-
 	// When ConcurrencyModel is specified but ContainerConcurrency
 	// is not (0), use the ConcurrencyModel value.
 	if rs.DeprecatedConcurrencyModel == RevisionRequestConcurrencyModelSingle && rs.ContainerConcurrency == 0 {
 		rs.ContainerConcurrency = 1
 	}
 
-	if rs.TimeoutSeconds == nil {
-		ts := cfg.Defaults.RevisionTimeoutSeconds
-		rs.TimeoutSeconds = &ts
-	}
-
-	c := rs.GetContainer()
-	if c.Resources.Requests == nil {
-		c.Resources.Requests = corev1.ResourceList{}
-	}
-	if _, ok := c.Resources.Requests[corev1.ResourceCPU]; !ok {
-		c.Resources.Requests[corev1.ResourceCPU] = cfg.Defaults.RevisionCPURequest
-	}
-
-	vms := c.VolumeMounts
-	for i := range vms {
-		vms[i].ReadOnly = true
+	if len(rs.Containers) == 0 {
```

Comment:
Comment here would be useful as it took me awhile to grok what is going on here and why it needs to happen

Reply:
Good call.